test/unit/number/parse.js

```diff
@@ -78,13 +78,18 @@ QUnit.test( "should parse zero-padded decimals", function( assert ) {
 	assert.equal( parse( "0.10", properties( "0.00", en ) ), 0.1 );
 });
 
+QUnit.test( "should parse non-padded decimals", function( assert ) {
+	assert.equal( parse( ".14159", properties( "0.0", en ) ), 0.14159 );
+	assert.equal( parse( ".752", properties( "0.0", en ) ), 0.752 );
+});
```

Comment:
Can you add tests for locales that use:
1. different decimal point (like Portuguese)?
1. different orientation (like Arabic)?

PS: You could format the number in these locales to know what the output would be and remove the `0`. :)